### PR TITLE
DD#0000: feat: Magento2: Add support for preload keys

### PR DESCRIPTION
**Pull request by valguss**
- Also updated to remove the config set and instead manually add the hocks to your deployer file
- And move var-exporter into the main composer require
- https://github.com/deployphp/deployer/pull/3453#issuecomment-1497248859

- [ - ] Bug fix https://github.com/deployphp/deployer/pull/3453#issuecomment-1497248859
- [ ] New feature?
- [ ] BC breaks?
- [ ] Tests added?
- [ - ] Docs added?

      Please, regenerate docs by running next command:
      $ php bin/docgen

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
         "symfony/polyfill-php80": "^1.22",
         "symfony/polyfill-php81": "^1.26",
         "symfony/process": "^5",
+        "symfony/var-exporter": "^5.4",
```
Comment (antonmedv) [inline composer.json:53]: Why is got here?

Comment (valguss) [reply composer.json:53]: Yeah I noticed i'd added it to the dev packages where infact I actually use it in the recipe so need it as part of the main packages

File: recipe/magento2.php [hunk 2]
```diff
@@ -443,7 +440,11 @@ function magentoDeployAssetsSplit(string $area)
 /**
  * Update cache id_prefix on deploy so that you are compiling against a fresh cache
  * Reference Issue: https://github.com/davidalger/capistrano-magento2/issues/151
- * use set('use_redis_cache_id') in your deployer script to enable
+ * use add the following to your deployer scripts:
```
Comment (schmengler) [inline recipe/magento2.php:443]: Grammar: "To use this feature, add the following to your deployer scripts:"

File: recipe/magento2.php [hunk 3]
```diff
@@ -454,6 +455,17 @@ function magentoDeployAssetsSplit(string $area)
     //set prefix to `alias_releasename_`
     $prefixUpdate = get('alias') . '_' . get('release_name') . '_';
 
+    //check for preload keys and update
+    if (isset($envConfigArray['cache']['frontend']['default']['backend_options']['preload_keys'])) {
+        $oldPrefix = $envConfigArray['cache']['frontend']['default']['id_prefix'];
+        $preloadKeys = $envConfigArray['cache']['frontend']['default']['backend_options']['preload_keys'];
+        $newPreloadKeys = [];
+        foreach ($preloadKeys as $preloadKey) {
+            $newPreloadKeys[] = str_replace($oldPrefix, $prefixUpdate, $preloadKey);
```
Comment (schmengler) [inline recipe/magento2.php:464]: to not accidentally replace parts of the cache keys if they match the old prefix, I would consider only replacing the prefix at the beginning of the string. Unlikely, I know, but using `substr_replace` or a regular expression seems more appropiate to me

**Review by schmengler — CHANGES_REQUESTED** [review]

File: recipe/magento2.php (hunk 2 above)
Comment (valguss) [reply recipe/magento2.php]: Thanks @schmengler 
That's now updated

File: recipe/magento2.php (hunk 3 above)
Comment (valguss) [reply recipe/magento2.php]: Thanks @schmengler 
That's now updated

File: recipe/magento2.php (hunk 2 above)
Comment (schmengler) [reply recipe/magento2.php]: "use add the following [...]" is still an incomplete sentence. If you want to keep it short: "Usage: add the following [...]"

File: recipe/magento2.php [hunk 4]
```diff
@@ -440,7 +440,7 @@ function magentoDeployAssetsSplit(string $area)
 /**
  * Update cache id_prefix on deploy so that you are compiling against a fresh cache
  * Reference Issue: https://github.com/davidalger/capistrano-magento2/issues/151
- * use add the following to your deployer scripts:
+ * To use this feature, add the following to your deployer scripts:
```
Comment (schmengler) [inline recipe/magento2.php:443]: forget my last comment, this is good now :smile:

**Review by schmengler — APPROVED** [review]

**Comment by antonmedv** [comment]
Ok for me. Let's get 👍🏻 from someone with Magento experience and merge.

**Comment by peterjaap** [comment]
@antonmedv 👍

**Review by schmengler — APPROVED** [review]

**Pull request merged**